packages/flutter_tools/lib/src/commands/screenshot.dart

```diff
@@ -123,16 +127,41 @@ class ScreenshotCommand extends FlutterCommand {
 
   Future<void> runScreenshot(File? outputFile) async {
     outputFile ??= globals.fsUtils.getUniqueFile(
-      globals.fs.currentDirectory,
+      fs.currentDirectory,
       'flutter',
       'png',
     );
+
+    // Additional error checks below for valid file name
+    // and directory because ios/xcode screenshot binary will
+    // return a 0 exit code even if outputFile is not valid
+
+    // Conditional for validating directory is valid
+    if (!fs.directory(outputFile.dirname).existsSync()) {
+      throwToolExit(
+        'The provided path to file needs to have a directory that exists\n'
+        'Directory: "${outputFile.dirname}" does not exist'
+      );
+    }
+
+    // Conditional for validating filename with allowed extensions
+    if (!RegExp(_kFileNamePattern).hasMatch(outputFile.basename)) {
+      throwToolExit(
+        'The provided filename is invalid\n'
+        'Allowed filename characters: "a-z", "A-Z", "0-9", "_", "-"\n'
+        'Allowed extensions: jpeg, jpg, png)\n'
```

Comment:
Might want to revisit this to determine what file types are supported through all targets